ludwig/automl/automl.py

```diff
@@ -0,0 +1,111 @@
+"""
+automl.py
+
+Driver script which:
+
+(1) Builds a base config by performing type inference and populating config
+    w/default combiner parameters, training paramers, and hyperopt search space
+(2) Tunes config based on resource constraints
+(3) Runs hyperparameter optimization experiment
+"""
+import logging
+import sys
+from typing import Dict, Union
+
+import numpy as np
+import pandas as pd
+from ludwig.automl.base_config import create_default_config
+from ludwig.hyperopt.run import hyperopt
+
+logger = logging.getLogger(__name__)
+
+
+try:
+    import dask.dataframe as dd
+    import ray
+except ImportError:
+    logger.error(
+        ' ray is not installed. '
+        'In order to use auto_train please run '
+        'pip install ludwig[ray]'
+    )
+    sys.exit(-1)
```

Comment:
I know we do this in a few other places in Ludwig, but for programatic usage, we should probably avoid calling `sys.exit` in case the user doesn't want their notebook to crash. Maybe raise an exception?

Reply:
Good point